cmd/resim/commands/workflows.go

```diff
@@ -642,6 +645,105 @@ func listWorkflowRuns(ccmd *cobra.Command, args []string) {
 	OutputJson(runs)
 }
 
+func workflowRunToSlackWebhookPayload(projectID uuid.UUID, workflowRun *api.WorkflowRun) *slack.WebhookMessage {
+	if workflowRun == nil || len(workflowRun.WorkflowRunTestSuites) == 0 {
+		// Return empty payload if no batches
+		return &slack.WebhookMessage{
+			Blocks: &slack.Blocks{BlockSet: []slack.Block{}},
+		}
+	}
+
+	var allBlocks []slack.Block
+	validBatchCount := 0
+
+	for _, suite := range workflowRun.WorkflowRunTestSuites {
+		// Skip if batch ID is nil UUID (disabled/skipped test suite)
+		if suite.BatchID == uuid.Nil {
+			continue
+		}
+
+		// Safely get the batch for this suite
+		batch, err := safeGetBatch(projectID, suite.BatchID)
+		if err != nil || batch == nil {
+			// Skip batches that don't exist (404) or other errors
+			continue
+		}
+
+		// Get Slack payload for this batch
+		batchPayload := batchToSlackWebhookPayload(batch)
+		if batchPayload == nil || batchPayload.Blocks == nil {
+			continue
+		}
+
+		// Add divider before each batch except the first valid one
+		if validBatchCount > 0 {
+			allBlocks = append(allBlocks, slack.NewDividerBlock())
+		}
+
+		// Add all blocks from this batch, making block_ids unique
+		for _, block := range batchPayload.Blocks.BlockSet {
+			// Make block_id unique by appending batch index
+			uniqueBlock := makeBlockIDUnique(block, validBatchCount)
+			allBlocks = append(allBlocks, uniqueBlock)
+		}
+
+		validBatchCount++
+	}
+
+	// Return empty payload if no valid batches were found
+	if validBatchCount == 0 {
+		return &slack.WebhookMessage{
+			Blocks: &slack.Blocks{BlockSet: []slack.Block{}},
+		}
+	}
+
+	return &slack.WebhookMessage{
+		Blocks: &slack.Blocks{BlockSet: allBlocks},
+	}
+}
+
+// safeGetBatch retrieves a batch and returns an error instead of calling log.Fatal
+func safeGetBatch(projectID uuid.UUID, batchID uuid.UUID) (*api.Batch, error) {
+	response, err := Client.GetBatchWithResponse(context.Background(), projectID, batchID)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve batch: %v", err)
+	}
+
+	// Use ValidateResponseSafe to handle 404s gracefully
+	if err := ValidateResponseSafe(http.StatusOK, "unable to retrieve batch", response.HTTPResponse, response.Body); err != nil {
+		return nil, err
+	}
+
+	return response.JSON200, nil
+}
+
+// makeBlockIDUnique modifies a block's block_id to be unique by appending an index
+func makeBlockIDUnique(block slack.Block, index int) slack.Block {
+	// Use type assertion to access block_id field
```

Comment:
suggestion: Instead of adding a new function to work around the existing slack integration, you could add a `idPrefix` or `idSuffix` parameter to `batchToSlackWebhookPayload` to mutate the IDs as they're constructed.   
  
You might even want to make that function just return a list of blocks, and create a wrapper function in `batch_summary.go` that actually wraps the blocks in the `WebhookMessage`.